### PR TITLE
Update CI to use clang-format 11.0.0.

### LINKED ISSUE #1975: Remove macos-10.15 due to GitHub Actions runner deprecation in 2022-12
GitHub Actions will deprecate the macos-10.15 runner in 2022-05 and remove it in 2022-12:
* https://github.com/actions/runner-images/issues/5583

This task is to remove the runner from our workflows.

### LINKED ISSUE #1976: Add macos-12.0 GitHub Actions to workflows
GitHub Actions will deprecate the macos-10.15 runner in 2022-05 and remove it in 2022-12, replacing it with macos-12.0:
* https://github.com/actions/runner-images/issues/5583

The #1975 task removes 10.15, and this current one adds 12.0.

### PR DESCRIPTION
**Pull request by ni4**
Commit 793e5b38 updates lint.yml for clang-format action, which started to use version 11.1.0 instead 11.0.0 by default, so this PR reverts this back to 11.0.0.
Also it updates macOS version runners.

@maxirmx  Was this done by intension, i.e. are there any benefits of switching to 11.1.0? I'm not against it but just want to get aligned/update work setup.

Fixes #1975 
Fixes #1976

### CODE REVIEW

**Comment by maxirmx** [comment]
> @maxirmx Was this done by intension, i.e. are there any benefits of switching to 11.1.0? I'm not against it but just want to get aligned/update work setup.

No, it was not done by intention

File: .github/workflows/macos.yml [hunk 1]
```diff
@@ -46,7 +46,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [macos-10.15, macos-11.0]
```
Comment (ronaldtse) [inline .github/workflows/macos.yml:49]: This change fixes:
* #1975
* #1976

**Comment by ronaldtse** [comment]
@ni4 if the macOS builds work then we can close both:
* #1975
* #1976

Thanks!

**Comment by ni4** [comment]
@ronaldtse yeah, let's just wait for the macOS 12 jobs to be picked up and handled.

**Comment by ni4** [comment]
Merging as this is just CI change and all is green.

**Pull request merged**